src/components/Questionnaire/QuestionnaireEditor.tsx

```diff
@@ -2012,24 +2014,50 @@ function QuestionEditor({
                   </CardContent>
                 ) : (
                   <CardContent className="space-y-4">
-                    <Autocomplete
-                      options={(valuesets ?? []).map((valueset) => ({
-                        label: valueset.name,
-                        value: valueset.slug,
-                      }))}
-                      value={
-                        question.answer_value_set === "valueset"
-                          ? ""
-                          : (question.answer_value_set ?? "")
-                      }
-                      onChange={(val: string) =>
-                        updateField("answer_value_set", val)
-                      }
-                      onSearch={setValueSetSearchQuery}
-                      placeholder={t("select_a_value_set")}
-                      isLoading={isFetchingValuesets}
-                      noOptionsMessage={t("no_valuesets_found")}
-                    />
+                    <div className="flex items-center gap-2 flex-col sm:flex-row">
+                      <div className="w-full">
+                        <Autocomplete
+                          options={(valuesets ?? []).map((valueset) => ({
+                            label: valueset.name,
+                            value: valueset.slug,
+                          }))}
+                          value={
+                            question.answer_value_set === "valueset"
+                              ? ""
+                              : (question.answer_value_set ?? "")
+                          }
+                          onChange={(val: string) =>
+                            updateField("answer_value_set", val)
+                          }
+                          onSearch={setValueSetSearchQuery}
+                          placeholder={t("select_a_value_set")}
+                          isLoading={isFetchingValuesets}
+                          noOptionsMessage={t("no_valuesets_found")}
+                        />
+                      </div>
+                      <Sheet open={isSheetOpen} onOpenChange={setIsSheetOpen}>
+                        <SheetTrigger asChild>
+                          <Button
+                            variant="outline"
+                            className="gap-2 w-full sm:w-auto"
+                          >
+                            <CareIcon icon="l-plus" />
+                            {t("create_valueset")}
+                          </Button>
+                        </SheetTrigger>
+                        <SheetContent
+                          side="right"
+                          className="w-full sm:max-w-2xl overflow-y-auto"
+                        >
+                          <ValueSetEditor
+                            onSuccess={(data) => {
+                              setIsSheetOpen(false);
+                              updateField("answer_value_set", data.slug);
```

Comment:
This will work if value set count < 14 and/or the newly created value set appears on first page.

You will need to handle if that isn't the case.